forge/forge/op/eval/forge/tm.py

```diff
@@ -1104,9 +1104,26 @@ def decompose(type, attr, dc, inputs):
             # We need to transpose to get the dimension we want to index by at the -2 position
             act = dc.op(TransposeTM.create(-2, dim), [act])
 
+        orig_act_shape = None
```

Comment:
We need to think a bit better about how to handle indexing on our end. For not, I'm okay to push this as is, but I'll create issues to explore these details in more detail. 

E.g. how to model and support different types of indexing through Forge-FE.

Reply:
Sure @nvukobratTT